src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -33,7 +33,16 @@ namespace Dynamo.ViewModels
 {
     public interface IDynamoViewModel : INotifyPropertyChanged
     {
-        ObservableCollection<WorkspaceViewModel> Workspaces { get; set; } 
+        ObservableCollection<WorkspaceViewModel> Workspaces { get; set; }
+
+        /// <summary>
+        /// Executes node and connector creation commands to create node1 and connect it to node2
+        /// </summary>
+        /// <param name="node1">input node to create</param>
+        /// <param name="node2">node to connect to</param>
+        /// <param name="portIndex1">output port of input node</param>
+        /// <param name="portIndex2">input node of given node</param>
+        void CreateAndConnectInputNode(NodeModel node1, NodeModel node2, int portIndex1, int portIndex2);
```

Comment:
I fear the `IDynamoViewModel` is not at all the place for this method. This should be on `ReadyParams`? 

Instead of adding methods like this in a piecemeal manner, why don't we expose Dynamo's recordable command framework? We would need to do this carefully, but this would be a much more general way to do this kind of thing.